# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/vr/XRInputSource.cpp
# Commit: ec6a64b639a8
# Full Hash: ec6a64b639a82e957376c09a71aef7ce62b3a5aa
# Author: Chris Martin <cmartin@mozilla.com>
# Date: 2020-12-03 09:47:26
# Regressor Bug: 1657404
# File Overlap Count: 3
# Description:
#   Bug 1657404 - Implement a strongly-typed, service-dependent gamepad handle r=handyman,aklotz
#   
#   Currently, the gamepad code uses a uint32_t as a handle and does some trickery
#   with it by trying to create a unique ID and adding an offset to it for VR code.
#   
# ==============================================================================

diff -r 6792f28b99bd -r ec6a64b639a8 dom/vr/XRInputSource.cpp
--- a/dom/vr/XRInputSource.cpp	Wed Dec 02 22:36:35 2020 +0000
+++ b/dom/vr/XRInputSource.cpp	Wed Dec 02 23:06:05 2020 +0000
@@ -198,12 +198,13 @@
   mTargetRaySpace = new XRInputSpace(aSession->GetParentObject(), aSession,
                                      nativeOriginTargetRay, aIndex);
 
-  const uint32_t gamepadId =
+  const uint32_t gamepadHandleValue =
       displayInfo.mDisplayID * gfx::kVRControllerMaxCount + aIndex;
-  const uint32_t hashKey = GamepadManager::GetGamepadIndexWithServiceType(
-      gamepadId, GamepadServiceType::VR);
+
+  const GamepadHandle gamepadHandle{gamepadHandleValue, GamepadHandleKind::VR};
+
   mGamepad =
-      new Gamepad(mParent, NS_ConvertASCIItoUTF16(""), -1, hashKey,
+      new Gamepad(mParent, NS_ConvertASCIItoUTF16(""), -1, gamepadHandle,
                   GamepadMappingType::Xr_standard, controllerState.hand,
                   displayInfo.mDisplayID, controllerState.numButtons,
                   controllerState.numAxes, controllerState.numHaptics, 0, 0);